src/olympia/promoted/models.py

```diff
@@ -0,0 +1,57 @@
+from django.db import models
+
+from olympia.addons.models import Addon
+from olympia.amo.models import ModelBase
+from olympia.constants.promoted import (
+    NOT_PROMOTED, PROMOTED_GROUPS, PROMOTED_GROUPS_BY_ID)
+from olympia.versions.models import Version
+
+
+class PromotedAddon(ModelBase):
+    GROUP_CHOICES = [(group.id, group.name) for group in PROMOTED_GROUPS]
+    group_id = models.SmallIntegerField(
+        choices=GROUP_CHOICES, default=NOT_PROMOTED.id)
+    addon = models.OneToOneField(
```

Comment:
This is what enforces the rule that any `addon` can have only one `PromotedAddon`?

Reply:
yep.  OneToOneField are inherently unique in django.